MSAL/src/MSALPublicClientApplication.m

```diff
@@ -249,31 +253,47 @@ - (id)initWithClientId:(NSString *)clientId
 {
     MSALAccountsProvider *request = [[MSALAccountsProvider alloc] initWithTokenCache:self.tokenCache
                                                                           clientId:self.clientId];
-    return [request allAccounts:error];
+    
+    NSArray *accounts = [request allAccounts:error];
+    if (error) *error = [MSALErrorConverter msalErrorFromMsidError:*error];
```

Comment:
nit: you might want to create another local error variable and convert it for clarity